bin/admin/benchmark_compare.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import subprocess
+import json
+import argparse
+
+SILENT_OUTPUT = False # Set to True to suppress command line output
+
+def run_command(command):
+    subprocess.run(command, shell=True, check=True, capture_output=SILENT_OUTPUT, text=True)
+
+# replace slashes in branch names with dashes for file naming, otherwise it will create directories
+def process_branch_name(branch_name):
+    return branch_name.replace('/', '-')
+
+def get_current_git_state():
+    try:
+        # try branch name
+        result = subprocess.run("git symbolic-ref --short HEAD", shell=True,
+                                capture_output=True, text=True, check=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError:
+        # else try commit hash (for detached state)
+        result = subprocess.run("git rev-parse HEAD", shell=True, check=True, capture_output=True, text=True)
+        return result.stdout.strip()
+
+def restore_git_state(original_git_state):
+    print(f"\nRestoring original git state: {original_git_state}")
+    run_command(f"git checkout {original_git_state}")
+
+def configure_and_build(commit, cmake_variables, benchmark_target, build_dir):
+    print(f"\nConfiguring and building commit: {commit}\n")
+
+    run_command(f"git checkout {commit}")
+
+    cmake_vars_str = " ".join(cmake_variables)
+    command = f"cmake -S . -B {build_dir} -DCMAKE_BUILD_TYPE=Release {cmake_vars_str}"
+    run_command(command)
+
+    command = f"cmake --build {build_dir} --target {benchmark_target} --clean-first"
+    run_command(command)
+
+def run_benchmarks(commit, benchmark_target, build_dir):
+    print(f"Running benchmarks for commit: {commit}\n")
+    output = process_branch_name(commit) + "-results.json"
+    command = f"./{build_dir}/benchmarks/{benchmark_target} --benchmark_out_format=json --benchmark_time_unit=us --benchmark_out={output}"
+    run_command(command)
+    print(f"Benchmarks for commit {commit} completed and results saved to {output}\n")
+
+def compare_benchmarks(base_benchmark, new_benchmark, metric="cpu_time"):
+    if metric not in ["cpu_time", "real_time"]:
+        raise ValueError("Invalid metric specified. Use 'cpu_time' or 'real_time'.")
```

Comment:
I would recommend using an enum for the metric. This will make it obvious which values are valid.